frontend/src/pages/Worksheet.js

```diff
@@ -101,17 +101,24 @@ function Worksheet() {
     setCourseModal([false, '']);
   }, []);
 
-  // Hide/Show this course
-  const toggleCourse = useCallback(
+  const updateHiddenCourses = useCallback(
     (crn) => {
       let new_hidden_courses = Object.assign({}, hidden_courses);
       if (hidden_courses[crn]) new_hidden_courses[crn] = false;
       else new_hidden_courses[crn] = true;
-      setHiddenCourses(new_hidden_courses);
+      return new_hidden_courses;
     },
     [hidden_courses]
   );
 
+  // Hide/Show this course
+  const toggleCourse = useCallback(
+    (crn) => {
+      setHiddenCourses(updateHiddenCourses(crn));
```

Comment:
Ah not quite what I meant originally - you can actually pass a function like this:
```javascript
setHiddenCourses((old_hidden_courses) => {
    // do transformation on old_hidden_courses
    return new_hidden_courses;
})
```

I just made the change myself because it's pretty quick.